app/controllers/collections_controller.rb

```diff
@@ -39,6 +42,61 @@ def show
     end
   end
 
+  def create
```

Comment:
The whole method is jvery big. Do you think you could decompose it into one main create method that calls a few helper methods?